src/src/com/microsoft/aad/adal/Discovery.java

```diff
@@ -189,13 +190,13 @@ private boolean sendRequest(final URL queryUrl) throws MalformedURLException, JS
         String errorCodes = "";
         try {
             ClientMetrics.INSTANCE.beginClientMetricsRecord(queryUrl, mCorrelationId, headers);
-            webResponse = mWebrequestHandler.sendGet(queryUrl, headers);
-            if (webResponse.getResponseException() == null) {
+            try {
+                webResponse = mWebrequestHandler.sendGet(queryUrl, headers);
                 ClientMetrics.INSTANCE.setLastError(null);
-            } else {
+            } catch (IOException e) {
                 ClientMetrics.INSTANCE.setLastError(String.valueOf(webResponse.getStatusCode()));
```

Comment:
I don't understand this ClientMetrics usage, so I'm not sure why we swallowing this exception...